### PR TITLE
fix(ipyMimeType): adding syntax highlighting for ipy extension

### LINKED ISSUE #2024: Syntax highlight in editor for *.ipy
Would it be possible for the editor to recognize `*.ipy` as an extension for python?
Right now `*.ipy` files are treated as pure text files, i.e. no syntax highlighting.

### PR DESCRIPTION
**Pull request by alexandercbooth**
Closes #2024.

### CODE REVIEW

File: packages/codemirror/src/mimetype.ts [hunk 1]
```diff
@@ -51,6 +51,9 @@ class CodeMirrorMimeTypeService implements IEditorMimeTypeService {
    * If a mime type cannot be found returns the defaul mime type `text/plain`, never `null`.
    */
   getMimeTypeByFilePath(path: string): string {
+    if (path.split('.').slice(-1)[0] === 'ipy') {
```
Comment (blink1073) [inline packages/codemirror/src/mimetype.ts]: Do you mind using [PathExt.extname](https://github.com/jupyterlab/jupyterlab/blob/77ac304d2633ce937c00e506b8e827e0c4de2f70/packages/coreutils/src/path.ts#L64) for this?

Comment (alexandercbooth) [inline packages/codemirror/src/mimetype.ts]: Absolutely!

Comment (alexandercbooth) [inline packages/codemirror/src/mimetype.ts]: Let me know if I didn't use it in the intended way and I can fix it up

**Comment by alexandercbooth** [comment]
Oops... Sorry, I'll fix this

File: packages/codemirror/src/mimetype.ts [hunk 2]
```diff
@@ -51,6 +51,9 @@ class CodeMirrorMimeTypeService implements IEditorMimeTypeService {
    * If a mime type cannot be found returns the defaul mime type `text/plain`, never `null`.
    */
   getMimeTypeByFilePath(path: string): string {
+    if (PathExt.extname === 'ipy') {
```
Comment (blink1073) [inline packages/codemirror/src/mimetype.ts]: oh, wait, this should be `PathExt.extname(path) === '.ipy'`

Comment (alexandercbooth) [inline packages/codemirror/src/mimetype.ts]: Ahh sorry, that was silly! Was staring at it for the last hour and overthinking it, thanks!!

Comment (blink1073) [inline packages/codemirror/src/mimetype.ts]: No worries, been there (frequently) myself.

**Comment by blink1073** [comment]
Thanks again!

**Comment by alexandercbooth** [comment]
My pleasure, thanks Steven!

**Pull request merged**